jib-cli/src/test/java/com/google/cloud/tools/jib/cli/cli2/JibCliTest.java

```diff
@@ -271,43 +373,42 @@ public void testParse_passwordWithoutUsername() {
   }
 
   @RunWith(Parameterized.class)
-  public static class UsernamePasswordNotAllowedWithToAndFrom {
-    @Parameterized.Parameters(name = "{0},{1}")
-    public static Collection<Object[]> data() {
+  public static class IncompatibleCredentialCombos {
+    @Parameterized.Parameters
+    public static Collection<String[][]> data() {
       return Arrays.asList(
-          new Object[][] {
-            {"--to-username", "--to-password"},
-            {"--from-username", "--from-password"}
+          new String[][][] {
+            {{"--credential-helper=x", "--to-credential-helper=x"}},
+            {{"--credential-helper=x", "--from-credential-helper=x"}},
+            {{"--credential-helper=x", "--username=x", "--password=x"}},
+            {{"--credential-helper=x", "--from-username=x", "--from-password=x"}},
+            {{"--credential-helper=x", "--to-username=x", "--to-password=x"}},
+            {{"--username=x", "--password=x", "--from-username=x", "--from-password=x"}},
+            {{"--username=x", "--password=x", "--to-username=x", "--to-password=x"}},
+            {{"--username=x", "--password=x", "--to-credential-helper=x"}},
+            {{"--username=x", "--password=x", "--from-credential-helper=x"}},
           });
     }
 
-    @Parameterized.Parameter(0)
-    public String usernameField;
-
-    @Parameterized.Parameter(1)
-    public String passwordField;
+    @Parameterized.Parameter public String[] authArgs;
+    private final String[] requiredArgs = new String[] {"--target=ignored"};
 
     @Test
     public void testParse_usernameWithoutPassword() {
+      try {
+        CommandLine.populateCommand(new JibCli(), ArrayUtils.addAll(requiredArgs, authArgs));
+      } catch (CommandLine.MaxValuesExceededException error) {
+        System.out.println("goose");
+      } catch (Exception ex) {
+        // do nothing
+      }
       MutuallyExclusiveArgsException meae =
           assertThrows(
               MutuallyExclusiveArgsException.class,
               () ->
                   CommandLine.populateCommand(
-                      new JibCli(),
-                      "--target",
-                      "test-image-ref",
-                      "--username",
-                      "test-username",
-                      "--password",
-                      "test-password",
-                      usernameField,
-                      "test-username",
-                      passwordField,
-                      "test-password"));
-      assertThat(meae.getMessage())
-          .isEqualTo(
-              "Error: [--username=<username> --password[=<password>]] and [[--to-username=<username> --to-password[=<password>]] [--from-username=<username> --from-password[=<password>]]] are mutually exclusive (specify only one)");
+                      new JibCli(), ArrayUtils.addAll(requiredArgs, authArgs)));
+      assertThat(meae.getMessage()).startsWith("Error: ");
```

Comment:
how about this?
```suggestion
      assertThat(meae).hasMessageThat().startsWith("Error:");
```
Might produce a more readable error message.